src/net/ossl.cc

```diff
@@ -1824,8 +1845,9 @@ const int seastar::tls::ERROR_UNKNOWN_SRP_USERNAME = ERR_PACK(
   ERR_LIB_SSL, 0, SSL_R_INVALID_SRP_USERNAME);
 const int seastar::tls::ERROR_PREMATURE_TERMINATION = ERR_PACK(
   ERR_LIB_SSL, 0, SSL_R_UNEXPECTED_EOF_WHILE_READING);
-const int seastar::tls::ERROR_PUSH = ERR_PACK(
-  ERR_LIB_SSL, 0, SSL_R_BIO_NOT_SET);
+// System errors are not ERR_PACK'ed like other errors but instead
+// are OR'ed with ((unsigned int)INT_MAX + 1)
+const int seastar::tls::ERROR_PUSH = int(ERR_SYSTEM_FLAG | EPIPE);
```

Comment:
question: can we ever see the old error `ERR_PACK(ERR_LIB_SSL, 0, SSL_R_BIO_NOT_SET)` as a push error or is that not possible anymore?

Also, just to double check, this is intentionally `int(ERR_SYSTEM_FLAG | EPIPE)` and not `ERR_PACK(ERR_LIB_???, 0, ERR_SYSTEM_FLAG | EPIPE)`, right?

Reply:
> question: can we ever see the old error ERR_PACK(ERR_LIB_SSL, 0, SSL_R_BIO_NOT_SET) as a push error or is that not possible anymore?

Fair question.  The push/pull errors are more artifacts of GnuTLS.  This was a best approximation I could do with what I think the `PULL` error on GnuTLS means with OpenSSL

> Also, just to double check, this is intentionally int(ERR_SYSTEM_FLAG | EPIPE) and not ERR_PACK(ERR_LIB_???, 0, ERR_SYSTEM_FLAG | EPIPE), right?

Yes.  When an `errno` is raised within OpenSSL, it is paired with `ERR_LIB_SYS` however OpenSSL drops the `lib` portion and just or's the errno with `ERR_SYSTEM_FLAG`

Reply:
Thanks, makes sense 👍